src/shared/components/post/post-listing.tsx

```diff
@@ -750,6 +752,25 @@ export class PostListing extends Component<PostListingProps, PostListingState> {
     );
   }
 
+  get markReadButton() {
+    const read = this.props.post_view.read;
+    const label = read ? i18n.t("mark read") : i18n.t("mark unread");
```

Comment:
Please don't put spaces in the translation strings. Use `mark_read` and `mark_unread` instead.